### PR TITLE
master

### PR DESCRIPTION
**Pull request by shachardon**
when splitting token with currency sign, tag it as unanalyzable 
(and not punctuation).
split tokens with apostrophe or hyphen according to an instruction file.

### CODE REVIEW

File: scripts/fix_tokenization.py [hunk 1]
```diff
@@ -42,6 +55,23 @@ def get_id(self):
         return ret
 
 
+def create_unit_element(state, text, tag):
+    elem = Element(SiteCfg.Tags.Terminal,
+                   {SiteCfg.Attr.SiteID: state.get_id()})
+    elem.text = text
+    preterminal_elem = Element(SiteCfg.Tags.Unit,
+                               {SiteCfg.Attr.SiteID: state.get_id()})
+    pretermoinal_parent = Element(SiteCfg.Tags.Unit,
```
Comment (danielhers) [inline scripts/fix_tokenization.py]: typo: should be `preterminal_parent`

File: scripts/fix_tokenization.py [hunk 2]
```diff
@@ -72,6 +102,23 @@ def insert_retokenized(terminal, preterminal_parent, tokens, index_in_pretermina
     insert_punct(insert_index + 1, preterminal_parent, state, tokens[non_punct_index + 1:])
 
 
+def insert_retokenized_currency(i, terminals, preterminals,
+                                preterminal_parents,
+                                tokens, state):
+    if len(tokens) == 2 and tokens[0] in CURRENCIES and \
+            (tokens[1].replace('.', '', 1).isdigit() or
+             tokens[1].replace(',', '')):
```
Comment (danielhers) [inline scripts/fix_tokenization.py]: What is the condition with the comma for? Shouldn't it also be with `isdigit`?

File: scripts/fix_tokenization.py [hunk 3]
```diff
@@ -72,6 +102,23 @@ def insert_retokenized(terminal, preterminal_parent, tokens, index_in_pretermina
...
+        terminals[i].text = tokens[1]
+        index_to_insert = preterminal_parents[i].getchildren(). \
+            index(preterminals[i])
+        preterminal_parents[i].insert(index_to_insert,
+                                      create_token_element(state,
+                                                           tokens[0],
+                                                           False))
```
Comment (danielhers) [inline scripts/fix_tokenization.py]: I think it's better to write `is_punct=False` instead of just `False` for clarity. Same for the similar cases below.

**Review by danielhers — CHANGES_REQUESTED** [review]
Looks good, but please see comments.

**Comment by shachardon** [comment]
Yes, it was supposed to be isdigit(),  I missed it when I did the merge.
Fixed all the comments.

‫בתאריך יום ב׳, 8 באוק׳ 2018 ב-15:35 מאת ‪Daniel Hershcovich‬‏ <‪
notifications@github.com‬‏>:‬

**Pull request merged**